esiclient/tests/v1/test_node_network.py

```diff
@@ -298,25 +299,59 @@ def test_take_action_port(self, mock_gfnifp):
         self.app.client_manager.baremetal.port.list.\
             return_value = [self.port1, self.port2]
 
-        arglist = ['node1', '--port', 'node1']
+        arglist = ['node1', '--port', 'node1-port']
         verifylist = []
 
         parsed_args = self.check_parser(self.cmd, arglist, verifylist)
 
         results = self.cmd.take_action(parsed_args)
         expected = (
             ["Node", "MAC Address", "Port", "Network", "Fixed IP"],
-            ["node1", "bb:bb:bb:bb:bb:bb", "node1", "test_network", "2.2.2.2"]
+            ["node1", "bb:bb:bb:bb:bb:bb", "node1-port", "test_network",
+             "2.2.2.2"]
         )
         self.assertEqual(expected, results)
         self.app.client_manager.network.find_port.\
-            assert_called_once_with("node1")
+            assert_called_once_with("node1-port")
         self.app.client_manager.network.get_network.\
             assert_called_once_with("network_uuid")
         self.app.client_manager.baremetal.node.vif_attach.\
             assert_called_once_with('node1', self.neutron_port.id)
         mock_gfnifp.assert_called_once
 
+    @mock.patch('esiclient.utils.get_full_network_info_from_port',
+                return_value=(["test_network"], ["2.2.2.2"]),
+                autospec=True)
+    def test_take_action_port_and_mac_address(self, mock_gfnifp):
+        self.app.client_manager.baremetal.node.get.\
+            return_value = self.node
+        self.app.client_manager.baremetal.port.get_by_address.\
+            return_value = self.port2
```

Comment:
Why port2?

Reply:
It's arbitrary between that and self.port1; since this is a unit test, we're just assuming that baremetal.port.get_by_address will return the correct value.